File: app/src/main/java/com/example/sarahn/waterreminderapp/classes/ClsJobService.java
package com.example.sarahn.waterreminderapp.classes;


import com.example.sarahn.waterreminderapp.Utils.Logging;
import com.example.sarahn.waterreminderapp.Utils.NotificationUtills;
import com.example.sarahn.waterreminderapp.Utils.SharedPrefUtils;
import com.firebase.jobdispatcher.JobParameters;
import com.firebase.jobdispatcher.JobService;


/**
 * Created by SarahN on 6/18/2017.
 */
public class ClsJobService extends JobService {

    public static JobParameters jobId;

    @Override
    public boolean onStartJob(JobParameters params) {

        if(SharedPrefUtils.getRequired(getApplicationContext()) == SharedPrefUtils.getConsumed(getApplicationContext())){
            Logging.logMessage("requirement is equal to consumed");

            return false;
        }else{

            NotificationUtills.notificationBuilder(getApplicationContext());
            Logging.logMessage("requirement is not equal to consumed");
            return true;
        }


    }

    @Override
    public boolean onStopJob(JobParameters params) {
        return false;
    }


}

File: app/src/main/java/com/example/sarahn/waterreminderapp/classes/ClsJobService.kt
package com.example.sarahn.waterreminderapp.classes

import com.example.sarahn.waterreminderapp.Utils.Logging
import com.example.sarahn.waterreminderapp.Utils.NotificationUtills
import com.example.sarahn.waterreminderapp.Utils.SharedPrefUtils
import com.firebase.jobdispatcher.JobParameters
import com.firebase.jobdispatcher.JobService

/**
 * Created by SarahN on 6/18/2017.
 */
class ClsJobService : JobService() {
    override fun onStartJob(params: JobParameters): Boolean {
        return if (SharedPrefUtils.getRequired(applicationContext) == SharedPrefUtils.getConsumed(applicationContext)) {
            Logging.logMessage("requirement is equal to consumed")
            false
        } else {
            NotificationUtills.notificationBuilder(applicationContext)
            Logging.logMessage("requirement is not equal to consumed")
            true
        }
    }

    override fun onStopJob(params: JobParameters): Boolean {
        return false
    }

    companion object {
        var jobId: JobParameters? = null
    }
}